core/src/main/java/com/klaytn/caver/tx/type/TxTypeSmartContractDeploy.java

```diff
@@ -88,4 +92,48 @@ public BigInteger getCodeFormat() {
     public Type getType() {
         return Type.SMART_CONTRACT_DEPLOY;
     }
+
+    /**
+     * decode transaction hash from sender to reconstruct transaction with fee payer signature.
+     *
+     * @param rawTransaction RLP-encoded signed transaction from sender
+     * @return TxTypeSmartContractDeploy decoded transaction
+     */
+    public static TxTypeSmartContractDeploy decodeFromRawTransaction(byte[] rawTransaction) {
+        // TxHashRLP = type + encode([nonce, gasPrice, gas, to, value, from, input, humanReadable, codeFormat, txSignatures])
+        try {
+            byte[] rawTransactionExceptType = KlayTransactionUtils.getRawTransactionNoType(rawTransaction);
+            RlpList rlpList = RlpDecoder.decode(rawTransactionExceptType);
+            List<RlpType> values = ((RlpList) rlpList.getValues().get(0)).getValues();
+
+            BigInteger nonce = ((RlpString) values.get(0)).asPositiveBigInteger();
+            BigInteger gasPrice = ((RlpString) values.get(1)).asPositiveBigInteger();
+            BigInteger gasLimit = ((RlpString) values.get(2)).asPositiveBigInteger();
+            //String to = ((RlpString) values.get(3)).asString();
```

Comment:
Why is this line processed as comment?

Reply:
We don't put the 'to address' in the smart contract because it is automatically generated. Therefore, there is 'to' in the encoded value and this value is not needed when decoding.  I've commented it out so you don't have to be confuse which of the values ​​you decoded.